### PR TITLE
perf/fix: selective header forwarding to user function 

### PR DESCRIPTION
**Pull request by johanandren**
List of headers to be forwarded in component options.

Manually verified against the updated proxy.

This is waiting on a release of proxy with lightbend/akkaserverless-framework#729 merged.

### CODE REVIEW

File: sdk/package-lock.json [hunk 1]
```diff
@@ -1,7093 +1,8 @@
 {
   "name": "@lightbend/akkaserverless-javascript-sdk",
   "version": "0.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (pvlugter) [inline sdk/package-lock.json:4]: The lock file was updated to v2 for npm 7 (in #19). We probably don't want to keep going back and forth, so using node 14 with upgraded npm 7 will be required locally now.

Comment (johanandren) [reply sdk/package-lock.json:4]: Ah, accidentally committed mostly

**Comment by andreaTP** [comment]
Not a big deal but I added some tests in #38 that can cover this use case:
https://github.com/lightbend/akkaserverless-javascript-sdk/pull/38/files#diff-a41d11ba5add430fe99954315fe0cf537270d4facfc493edf1abb84537af6ba6R144-R167

**Comment by johanandren** [comment]
Note that this still is waiting for the next proxy release with the protocol change.

File: sdk/test/akkaserverless-test.js [hunk 2]
```diff
@@ -15,6 +15,7 @@
  */
 
 const should = require('chai').should();
+const expect = require('chai').expect;
```
Comment (andreaTP) [inline sdk/test/akkaserverless-test.js:18]: should we converge to one testing style?
I am not strongly opinionated but I do prefer to stitch with one "per-cdebase"

File: sdk/package-lock.json (hunk 1 above)
Comment (andreaTP) [reply sdk/package-lock.json:4]: still on `v1`

File: sdk/test/akkaserverless-test.js [hunk 3]
```diff
@@ -154,6 +156,9 @@ At package.test.json:2:4:
     comp.getComponentType().should.equal('my-type');
     comp.getEntity().getEntityType().should.equal('my-entity-type');
     should.equal(comp.getEntity().getPassivationStrategy(), undefined);
+    expect(comp.getEntity().getForwardHeadersList()).to.have.same.members([
```
Comment (andreaTP) [inline sdk/test/akkaserverless-test.js:159]: should we have a quick test for the two possible code-paths?

File: sdk/test/akkaserverless-test.js (hunk 2 above)
Comment (johanandren) [reply sdk/test/akkaserverless-test.js:18]: Sure, no strong preference, just didn't find API to express the same assertion with `should`, do you know how that would look for the this array should have the same elements as that array?

Comment (pvlugter) [reply sdk/test/akkaserverless-test.js:18]: Should be able to just rewrite in `should` format:

```js
comp.getEntity().getForwardHeadersList().should.have.same.members([
  'x-my-header'
]);
```

File: sdk/test/akkaserverless-test.js (hunk 3 above)
Comment (andreaTP) [reply sdk/test/akkaserverless-test.js:159]: yup, is it supposed to happen at all?

Comment (johanandren) [reply sdk/test/akkaserverless-test.js]: Yeah, action should be able to pick up headers as well

Comment (johanandren) [reply sdk/test/akkaserverless-test.js]: Covered

**Review by pvlugter — APPROVED** [review]
LGTM

**Pull request merged**